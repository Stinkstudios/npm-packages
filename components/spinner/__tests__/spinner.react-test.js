import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import Spinner from '../spinner';

const component = () => <Spinner size={50} />;

it('renders DOM output correctly', () => {
  const tree = renderer.create(component()).toJSON();

  expect(tree).toMatchSnapshot();
});

it("changes component's size when using 'size' prop", () => {
  const tree = mount(<Spinner size={150} />);

  const topLevel = tree.first();

  expect(topLevel.props().size).toBe(150);
  expect(topLevel.find('.spinner__inner').get(0).style._values.width).toBe(
    '150px'
  );
  expect(topLevel.find('.spinner__inner').get(0).style._values.height).toBe(
    '150px'
  );
});
