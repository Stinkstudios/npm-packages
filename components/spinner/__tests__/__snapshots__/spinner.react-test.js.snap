// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders DOM output correctly 1`] = `
<div
  className="spinner"
>
  <div
    className="spinner__inner"
    style={
      Object {
        "height": 50,
        "width": 50,
      }
    }
  >
    <div
      className="spinner__half spinner__half--left"
    />
    <div
      className="spinner__half spinner__half--right"
    />
  </div>
</div>
`;
